### DISCUSSION TITLE
add metadata/details into EXE-files, plz.

### DISCUSSION THREAD

**Discussion by zedd-0** [rustdesk/rustdesk#5533] [Feature Request]
hi,
unfortunately, i cannot identify the version number of the EXE-files,
because no metadata was specified for it.
could you integrate this into the EXE-files in the future?

```
PS E:\TEMP> (Get-ChildItem 'rustdesk-1.2.3-x86_64.exe').VersionInfo | select *


FileVersionRaw     : 0.0.0.0
ProductVersionRaw  : 0.0.0.0
Comments           :
CompanyName        :
FileBuildPart      : 0
FileDescription    :
FileMajorPart      : 0
FileMinorPart      : 0
FileName           : E:\TEMP\rustdesk-1.2.3-x86_64.exe
FilePrivatePart    : 0
FileVersion        :
InternalName       :
IsDebug            : False
IsPatched          : False
IsPrivateBuild     : False
IsPreRelease       : False
IsSpecialBuild     : False
Language           :
LegalCopyright     :
LegalTrademarks    :
OriginalFilename   :
PrivateBuild       :
ProductBuildPart   : 0
ProductMajorPart   : 0
ProductMinorPart   : 0
ProductName        :
ProductPrivatePart : 0
ProductVersion     :
SpecialBuild       :



PS E:\TEMP>
```

**Comment by xlionjuan** [comment]
I guess you watched ThioJoe's video about AppLocker.

**Comment by zedd-0** [comment]
sorry,
but no, i did not watched ThioJoe's video about AppLocker.
i'm not a native speaker.
but for security reasons it is always good.

I personally need it for version identification and renaming for my download script.
the last big rename after v1.1.9 did completely broke it.

:D

**Reply by xlionjuan** [reply]
Try GitHub API to get the latest release

**Reply by zedd-0** [reply]
i already use the GitHub API,
but the nightly version has no version-tag.
version numbers inside of a filename is not very accurate.

**Reply by xlionjuan** [reply]
create cron jobs to run it everyday

**Comment by deep-soft** [comment]
> 

You're trying to extract version information from self-extracting executable.
Only rustdesk.exe from installation folder have this information.
You have to extract the self-extracting executable, search for rustdesk.exe and apply that powershell script on rustdesk.exe.

**Reply by zedd-0** [reply]
i cannot extract it with winrar/7-zip.

i cannot extract the x86_64-file on a x86 32bit system.
how can i extract it on the command line?

it would be nice if you can sign the installer, too, please.

**Reply by deep-soft** [reply]
self-extracting executable it's created with brotli.
7zip and winrar can't extract them.
Of course the easiest way is to include version information in self-extracting executable.
Sign the installer is easy, can be done in yml workflow but adding the version information to executable ...

**Comment by deep-soft** [comment]
[windows flutter - include version information in self-extracted executable](https://github.com/rustdesk/rustdesk/discussions/5560)

https://github.com/rustdesk/rustdesk/discussions/5560#discussion-5578928

**Comment by deep-soft** [comment]
@zedd-0 
https://github.com/rustdesk/rustdesk/releases/download/nightly/rustdesk-1.2.3-x86_64.exe

**Reply by zedd-0** [reply]
works fine.
thanks a lot.

```
(Get-ChildItem .\rustdesk-1.2.3-x86_64.exe).VersionInfo | select *

FileVersionRaw     : 1.2.3.39
ProductVersionRaw  : 1.2.3.39
Comments           :
CompanyName        : com.carriez
FileBuildPart      : 3
FileDescription    : rustdesk
FileMajorPart      : 1
FileMinorPart      : 2
FileName           : D:\.Users\MiC\Downloads\rustdesk-1.2.3-x86_64.exe
FilePrivatePart    : 39
FileVersion        : 1.2.3+39
InternalName       : rustdesk
IsDebug            : False
IsPatched          : False
IsPrivateBuild     : False
IsPreRelease       : False
IsSpecialBuild     : False
Language           : Englisch (Vereinigte Staaten)
LegalCopyright     : Copyright (C) 2023 com.carriez. All rights reserved.
LegalTrademarks    :
OriginalFilename   : rustdesk.exe
PrivateBuild       :
ProductBuildPart   : 3
ProductMajorPart   : 1
ProductMinorPart   : 2
ProductName        : rustdesk
ProductPrivatePart : 39
ProductVersion     : 1.2.3+39
SpecialBuild       : 
```
